From dd7d8b358930b07ab4047325fb27ce1da5510486 Mon Sep 17 00:00:00 2001
From: Konstantin Ozerkov <kozerkov@openvz.org>
Date: Thu, 13 Nov 2008 14:33:09 +0300
Subject: [PATCH 159/159] Fix OOPS while stopping VE after binfmt_misc.ko loaded

ve_binfmt_fini() should check if current VE have registered binfmt_misc fs.
(Properly handling situation while stopping VE which started before
 binfmt_misc.ko loaded)

http://bugzilla.openvz.org/show_bug.cgi?id=1028

Singed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
---
 fs/binfmt_misc.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/fs/binfmt_misc.c b/fs/binfmt_misc.c
index b91aaab..5bc2154 100644
--- a/fs/binfmt_misc.c
+++ b/fs/binfmt_misc.c
@@ -800,6 +800,8 @@ static void ve_binfmt_fini(void *x)
 	 * no locks since exec_ve is dead and noone will
 	 * mess with bm_xxx fields any longer
 	 */
+	if (!ve->bm_fs_type)
+		return;
 	dm_genocide(ve);
 	unregister_ve_fs_type(ve->bm_fs_type, NULL);
 }
-- 
1.5.4.3

